### PR TITLE
Apply awaitility matchers for other instance messages MODINVSTOR-1001

### PR DESCRIPTION
**Pull request by marcjohnson-kint**
Follow on from #862 

Applies a similar approach to previous changes by using Hamcrest matchers directly with awaitility to avoid catching assertion errors or relying on ordering of messages.

The checks for multiple created messages uses a check for the expected number of messages and a loop instead of a combined matcher because the use of the combined matcher was unstable on Jenkins, yet stable locally.

### CODE REVIEW

File: src/test/java/org/folio/rest/support/messages/InstanceEventMessageChecks.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import io.vertx.core.json.JsonObject;
+
+public class InstanceEventMessageChecks {
+  private static final EventMessageMatchers eventMessageMatchers
+    = new EventMessageMatchers(TENANT_ID, vertxUrl(""));
+
+  private InstanceEventMessageChecks() { }
+
+  public static void noInstanceMessagesPublished(String instanceId) {
+    await().during(1, SECONDS)
```
[the remark this answers is not in the document]
Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/support/messages/InstanceEventMessageChecks.java:29]: As I understand it, this uses the `await` method from the domain event assertions class that already defines an `atMost`. Does that not apply?

[the remark this answers is not in the document]
Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/support/messages/InstanceEventMessageChecks.java]: It is not intended. Whilst I think this is unlikely to occur in practice, I agree with you that it should not be permissible in these tests.

I've changed the code to include an `at most` constraint that matches the `during` constraint.

[the remark this answers is not in the document]
Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/support/messages/InstanceEventMessageChecks.java]: Hmm, it seems that introducing the `at most` with the same duration as `during` has made the tests unstable.

I've moved this PR to draft until that is resolved.

[the remark this answers is not in the document]
Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/support/messages/InstanceEventMessageChecks.java]: It seems using the same value for both is a known failure situation - https://stackoverflow.com/questions/62830176/unpredictable-behavior-around-during-and-atmost-of-awaitility

**Review by wwelling — CHANGES_REQUESTED** [review]
LGTM

**Comment by marcjohnson-kint** [comment]
> Rewrite code to be testable without race conditions.

What code are you suggesting should be rewrittten?

**Comment by wwelling** [comment]
> > Rewrite code to be testable without race conditions.
> 
> What code are you suggesting should be rewrittten?

That I do not know, as no convention is in place to easily map between test and code being tested.

**Pull request merged**